l5kit/l5kit/evaluation/extract_metric.py

```diff
@@ -0,0 +1,59 @@
+import numpy as np
+
+from . import read_gt_csv, read_pred_csv
+from .metrics import neg_multi_log_likelihood
+
+
+def compute_error_csv(ground_truth_path: str, inference_output_path: str) -> np.ndarray:
+    """
+    Arguments:
+        ground_truth_path (str): Path to the ground truth csv file.
+        inference_output_path (str): Path to the csv file containing network output.
+    """
+    ground_truth = {}
+    inference = {}
+
+    for el in read_gt_csv(ground_truth_path):
+        ground_truth[el["track_id"] + el["timestamp"]] = el
+    for el in read_pred_csv(inference_output_path):
+        inference[el["track_id"] + el["timestamp"]] = el
+
+    def validate(ground_truth: dict, inference: dict) -> bool:
+        valid = True
+
+        if not (len(ground_truth.keys()) == len(inference.keys())):
+            print(
+                f"""Incorrect number of rows in inference csv. Expected {len(ground_truth.keys())},
+                Got {len(inference.keys())}"""
+            )
+            valid = False
+
+        missing_obstacles = ground_truth.keys() - inference.keys()
```

Comment:
Why is it called obstacles - these are missing / unpredicted agents rights?

Reply:
yeah good catch..let me refactor this old function